### DISCUSSION TITLE
не все data source понимает

### DISCUSSION THREAD

**Discussion by lvPokka** [ooflet/Mi-Create#22] [Вопросы-RU]
Добрый день,

1)
unpack.exe не может скушать половины циферблатов с китайского форума www.bandbbs.cn из ветки mi band 9.
Хотя он детектить mi band 8 циферблат

Watch detected: MiBand8
System.DivideByZeroException: Attempted to divide by zero.
   at UnpackMiColorFace.Helpers.BmpHelper.AlignRowData(Byte[] src, Int32 width, Int32 height, Int32 type, Int32& dstWidth)

Вот пример
[bad-wface.zip](https://github.com/user-attachments/files/16482672/bad-wface.zip)

2) Есть циферблат с кучей данных, которая не видит ваша программа 
Data source not found.

Вот пример
[missing-wface.zip](https://github.com/user-attachments/files/16482708/missing-wface.zip)

**Comment by ooflet** [comment]
1. Я не разрабатываю декомпилятор, обратитесь за помощью к m0tral.
2. «Data source not found.» скорее всего означает, что источник данных был установлен с помощью более старой версии декомпилятора или EasyFace. С тех пор идентификаторы источников данных изменились
